consensus-engine/src/overlay/membership.rs

```diff
@@ -0,0 +1,42 @@
+// std
+
+// crates
+use rand::prelude::{SliceRandom, StdRng};
+use rand::SeedableRng;
+
+// internal
+use crate::overlay::CommitteeMembership;
+use crate::NodeId;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct FisherYatesShuffle {
+    entropy: [u8; 32],
+}
+
+impl FisherYatesShuffle {
+    pub fn new(entropy: [u8; 32]) -> Self {
+        Self { entropy }
+    }
+
+    pub fn shuffle<T: Clone>(elements: &[T], entropy: [u8; 32]) -> Vec<T> {
+        let mut elements = elements.to_vec();
+        let mut rng = StdRng::from_seed(entropy);
+        elements.shuffle(&mut rng);
+        elements
+    }
+}
+
+impl CommitteeMembership for FisherYatesShuffle {
+    fn reshape_committees(&self, nodes: &[NodeId]) -> Vec<NodeId> {
+        FisherYatesShuffle::shuffle(nodes, self.entropy)
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct FreezeMembership;
+
+impl CommitteeMembership for FreezeMembership {
+    fn reshape_committees(&self, nodes: &[NodeId]) -> Vec<NodeId> {
```

Comment:
The thing is that we would have to clone it before calling here because we retain old state 🤔 I'll check, maybe is worth it anyway. Thanks!

Reply:
> The thing is that we would have to clone it before calling here because we retain old state

That's what I was thinking, but anyway I think it might be better to have the clone there as it's clear why it's needed. Also, we might want to use this is a setting where we don't care about the old state

Reply:
>  Also, we might want to use this is a setting where we don't care about the old state

Absolutely, will change it! :)